File: frontend/src/types/prediction.ts
export type PredictionRequest = {
  area_sqft: number;
  floor_num: number;
  bathroom_num: number;
  balcony_num: number;
  location: string;
  furnishing: string;
  transaction: string;
  ownership: string;
  facing: string;
};

export type PredictionResponse = {
  predicted_price: number;
};

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class PredictionRequest(BaseModel):
    location: str = Field(..., min_length=1)
    area_sqft: float = Field(..., gt=0)
    floor_num: float
    bathroom_num: float = Field(..., ge=0)
    balcony_num: float = Field(..., ge=0)
    furnishing: str = Field(..., min_length=1)
    transaction: str = Field(..., min_length=1)
    ownership: str = Field(..., min_length=1)
    facing: str = Field(..., min_length=1)


class PredictionResponse(BaseModel):
    predicted_price: float